src/components/flag/code/templates/FlagComponentTemplate.jsx

```diff
@@ -3,21 +3,12 @@ var classNames = require('classnames');
 
 module.exports = function(props) {
   var classes = {
-    'ui-component-flag': true,
-    'flag-left': props.left,
-    'flag-right': props.right,
-    'flag-top': props.top,
-    'flag-bottom': props.bottom,
-    'flag-primary': props.primary,
-    'flag-secondary': props.secondary,
-    'flag-success': props.success,
-    'flag-info': props.info,
-    'flag-warning': props.warning,
-    'flag-danger': props.danger,
-    'flag-sm': props.small,
-    'flag-md': props.medium,
-    'flag-lg': props.large
+    'component-flag': true,
   };
+  if(props.size) classes[props.size] = true;
+  if(props.purpose) classes[props.purpose] = true
+  if(props.position) classes[props.position] = true
```

Comment:
@joestump 

Thanks for the well structured argument, makes a lot of sense. I don't have a well reasoned argument to oppose yours, only that our loose style guides that we started a few years ago were based on the [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript#blocks), which listed one line `if`s as "good". Happy to go with you on this project :)